Source/Core/PolygonGeometry.js

```diff
@@ -661,30 +673,48 @@ define([
         var results = PolygonGeometryLibrary.polygonsFromHierarchy(polygonHierarchy);
         var hierarchy = results.hierarchy;
         var polygons = results.polygons;
+        var i;
 
         if (polygons.length === 0) {
             return undefined;
         }
 
-        outerPositions = polygons[0];
+        outerPositions = polygons[0].slice();
```

Comment:
Ouch, between this and the scale below, we are definitely not following the C++ motto of "pay for only what you use", instead, we are slowing down and creating GC pressure for the most common case (polygon on the ground) where this is not needed.  Is there a reasonable why to avoid this?  Even if it means introducing a flag to say the polygon has heights?